packages/pigeon/lib/swift_generator.dart

```diff
@@ -650,14 +659,10 @@ if (wrapped == nil) {
     assert(!type.isVoid);
     if (type.baseName == 'Object') {
       return value + (type.isNullable ? '' : '!');
-    } else if (type.baseName == 'int') {
-      if (type.isNullable) {
-        // Nullable ints need to check for NSNull, and Int32 before casting can be done safely.
-        // This nested ternary is a necessary evil to avoid less efficient conversions.
-        return 'isNullish($value) ? nil : ($value is Int64? ? $value as! Int64? : Int64($value as! Int32))';
-      } else {
-        return '$value is Int64 ? $value as! Int64 : Int64($value as! Int32)';
-      }
+      // There's a swift bug that breaks force-casting nullable enums in maps.
```

Comment:
That's about all I know about it. It being a bug is an assumption based on the fact that the behavior is inconsistent with every other type.

Reply:
Rather than say it's a bug since we don't know that for sure, let's describe the behavior. E.g., "Enums require special handling, because unlike all other types `as!` casts to [I forget the details but explain the cast and the exception]"